linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java

```diff
@@ -99,6 +99,29 @@ public Object evaluate(Evaluator evaluator) {
       default:
         throw cannotEvaluate();
       }
+    case Mod:
+      switch (primitive) {
```

Comment:
You may also define the reference in RexImpTable[1].

[1] https://github.com/apache/calcite/blob/2dc97e6723e1b5bf762540f87ffffb5cd1a848a1/core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java#L338